### DISCUSSION TITLE
日志事件过滤 & 动作触发相关

### DISCUSSION THREAD

**Discussion by gito10** [Openwrt-Passwall/openwrt-passwall#3140] [Show and tell]
新增的日志事件过滤，暂时只支持Socks自动切换的日志，根据自己需要监测的内容，设置合适的正则表达式来筛选特定的日志。当满足筛选条件的日志事件发生时，会执行设定的shell 命令，可以直接设置命令，也可以设置脚本文件。
过滤器命中的日志内容，用%s代替。

下面会慢慢增加一些实例，为防部分用户不会用，简单说明下：
1. 在路由器某个目录（比如`/root`）新建一个 **.sh** 后缀的文件，比如AnPush的就叫 **anpush.sh**，Bark的就叫 **bark.sh**，或者随意；
2. 复制示例中给出的脚本内容，粘贴到新建的文件中，并将脚本开头列出的一些变量填入/修改为自己的值，然后保存；
3. 赋予脚本可执行权限(+x)，使用GUI软件操作或者Shell 命令 `chmd +x /path/to/script.sh`。
4. 在 Passwall 首页->日志，勾选高级日志，填上最后两个选项：
  ***日志事件过滤器*** 填上需要筛选的关键词或者正则表达式；
  ***Shell 命令*** 填上脚本路径，后面空格再加上一个 **%s**。比如脚本路径为 **/root/example.sh**，那么就填`/root/example.sh %s`。

如果下面实例中目前没有所需的，有能力动手的，可以自己实现或者照着现有的脚本代码自己修改适配，实现成功了，可以把脚本发上来。
自己不会改的，需要别人帮忙适配的，直接在这个Comment 下面回复，不要单开一个Comment。

**Comment by XedGo** [comment]
老哥，实例在哪了，急用！！！！

**Reply by gito10** [reply]
已更新 **AnPush 推送** 和 **Bark 推送**，不知道你是想用哪个推送服务，或者想执行其他的非推送动作。

不过这个功能只是刚提交，还没发布新版，你自己手动编译的吗？

**Comment by gito10** [comment]
# AnPush 推送
AnPush 支持 GET 和 POST 两种请求，因为请求内容也不复杂，模板直接就使用GET来请求了。
```shell
token=    #填入自己的Token
channel=  #填入推送通道ID
title="Passwall Notification" #推送标题，支持空格，后面有处理，但是符号应该不行，因为没有进行URL转码

#转码标题空格
title=$(echo -n $title | sed "s/ /+/g")
#日志内容URL转码
content=$(echo -n $@ | curl -Gso /dev/null -w %{url_effective} --data-urlencode @- "" | cut -c 3-)
#CURL GET 请求
curl --location --request GET "https://api.anpush.com/push/$token?title=$title&content=$content&channel=$channel"
```

**Comment by gito10** [comment]
# Bark 推送
Bark 跟 AnPush 差不多，也是支持 GET 和 POST 两种请求，而且请求 URL 更简单，但是 Bark 开源，支持自己搭建服务，所以多一个服务器选项。
```shell
server=api.day.app #服务器，如果自己有搭建，修改为自己的服务器地址
key=               #填入自己的Key
title="Passwall Notification" #推送标题，支持空格，后面有处理，但是符号应该不行，因为没有进行URL转码

#转码标题空格
title=$(echo -n $title | sed "s/ /+/g")
#日志内容URL转码
content=$(echo -n $@ | curl -Gso /dev/null -w %{url_effective} --data-urlencode @- "" | cut -c 3-)
#CURL GET 请求
curl --location --request GET "https://$server/$key/$title/$content"
```

**Reply by gito10** [reply]
@kxdn3 试试行不行。
AnPush的title是在参数部分，bark是在路径里面，你看这个标题里的空格转成“+”号能不能正常显示，还是需要转成“%20”才行。

还有Bark推送服务你是自己搭建的，还是用的官方的服务器？
